python/graphstorm/graphpartitioning/metadata_schema.py

```diff
@@ -0,0 +1,799 @@
+            the input graph dataset
+        """
+        if file_name is None or not isinstance(file_name, str):
+            raise ValueError(f"File path is not valid. = {file_name}")
+
+        self._basedir = None
+        if base_dir is not None and os.path.isdir(base_dir):
+            if not os.path.isfile(os.path.join(base_dir, file_name)):
+                raise ValueError(
+                    f"Metadata file does not exist. {base_dir}, {file_name}"
+                )
+            self._metadata_path = os.path.join(
+                os.path.join(base_dir, file_name)
+            )
+            self._basedir = base_dir
+        else:
+            if not os.path.isfile(file_name):
+                raise ValueError(f"Metadata file does not exist. {file_name}")
+            self._metadata_path = file_name
+
+        if self._basedir is None:
+            self._basedir = os.getcwd()
+
+        with open(self._metadata_path, "r", encoding="utf-8") as handle:
+            self._data = json.load(handle)
+
+    def _check_ifexists(self, data_files, src_type):
+        """Helper function to verify the presence of a set of filenames.
+
+        Argument
+        --------
+        data_files: list of str
+            A set of file names, either in absolute or in relative formats.
+        src_type: str
+            Either node type or edge type from the input graph.
+        """
+        if not isinstance(data_files, list):
+            raise ValueError(
+                f"Incorrect value encountered when iterating over files for {src_type}."
+            )
+
+        for filename in data_files:
+            if not os.path.isabs(filename):
+                if not os.path.isfile(os.path.join(self._basedir, filename)):
+                    raise FileNotFoundError(
+                        f"File: {os.path.join(self._basedir, filename)} does not exist"
+                    )
+            else:
+                if not os.path.isfile(filename):
+                    raise FileNotFoundError(f"File: {filename} does not exist.")
+
+    def init(self, file_path, base_dir, init_maps=True):
+        """Initialization function of this class to create all the data
+        structures used during typical processing of this pipeline.
+
+        Argument
+        --------
+        file_path: str
+            Path of the metadata schema file of the input dataset. It can be
+            either in absolute or relative formats.
+        base_dir: str
+            Absolute path of the directory in which the metadata schema file
+            is expected to be present. If not provided then current workding
+            directory is used instead to locate the metadata schema file.
+        init_maps: bool
+            Boolean flag to indicate whether to build all the necessary data
+            structures during initialization. Otherwise the user of this class
+            is exptected to invoke the appropriate initialization function.
```

Comment:
In general what's the value of having this option? Isn't it expected that we should init all the maps every time?

```suggestion
        init_maps: bool
            Boolean flag to indicate whether to build all the necessary data
            structures during initialization. Otherwise the user of this class
            is expected to invoke the appropriate initialization function.
```

Reply:
As described earlier, this was created so that unit tests could be performed with minimal setup.

Reply:
Is the only reason this exists to make unit tests easier?

Reply:
Yes. This is the only reason they are present. To enable modular tests with minimal setup.